### PR TITLE
Remove `$db` property from constructor

### PR DESCRIPTION
**Pull request by Tigrov**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #364

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,26 +106,44 @@ final class User extends ActiveRecord
...
+use Yiisoft\Db\Connection\ConnectionInterface;
 
 final class Register
 {
     public function register(
-        User $user
+        ConnectionInterface $db,
     ): ResponseInterface {
-        /** Connected AR by di autowired. */
+        ConnectionProvider::set($db);
```
Comment (samdark) [inline README.md:122]: Should it be set somewhere in bootstrap or config?

File: src/ConnectionProvider.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    /**
+     * Returns a connection by name.
+     */
+    public static function get(string $name = self::DEFAULT): ConnectionInterface
+    {
+        return self::$connections[$name];
+    }
+
+    /**
+     * Checks if a connection name exists.
```
Comment (samdark) [inline src/ConnectionProvider.php:38]:
```suggestion
     * Checks if a connection with the given name exists.
```

File: src/ConnectionProviderMiddleware.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Yiisoft\ActiveRecord;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Db\Connection\ConnectionInterface;
+
+class ConnectionProviderMiddleware implements MiddlewareInterface
```
Comment (samdark) [inline src/ConnectionProviderMiddleware.php:13]: Where is it supposed to be used?

Comment (Tigrov) [reply src/ConnectionProviderMiddleware.php:13]: Mentioned this in the doc

https://github.com/yiisoft/active-record/pull/368/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R132-R146

File: README.md (hunk 1 above)
Comment (Tigrov) [reply README.md:122]: This is an example with direct usage. And there is example with `ConnectionProviderMiddleware` below.
Also added information on how else middleware can be configured.

**Review by samdark — APPROVED** [review]

**Pull request merged**